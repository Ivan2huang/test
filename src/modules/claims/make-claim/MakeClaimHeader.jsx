import React from 'react';
import { injectIntl } from 'react-intl';
import PropTypes from 'prop-types';

import { Box } from '@material-ui/core';

import Typography from '../../../uiComponents/Typography';
import { formatMessage } from '../../../helpers/helpers';

const MakeClaimHeader = ({ intl }) => (
  <Box
    mb={{
      xs: 8,
      sm: 15,
    }}
  >
    <Box>
      <Typography type="style-1">
        {formatMessage(intl, 'claim.makeClaim.header.title', 'Make a claim')}
      </Typography>
    </Box>
    <Box mt={3}>
      <Typography type="style-8">
        {formatMessage(
          intl,
          'claim.makeClaim.header.description',
          'Claim submission is only applicable to designated outpatient ' +
            'consultation incurred in HK (refer to ‘Consultation Type’ in the next step). ' +
            'Please keep your original receipt as HSBC Life BenefitsPlus reserves the right to request for ' +
            'the original receipt within 6 months of the claim completion date. ' +
            'For details, please refer to the leaflets.',
        )}
      </Typography>
    </Box>
  </Box>
);

MakeClaimHeader.propTypes = {
  intl: PropTypes.shape({}).isRequired,
};

export default injectIntl(MakeClaimHeader);
