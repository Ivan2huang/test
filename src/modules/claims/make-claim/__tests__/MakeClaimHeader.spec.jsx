/* eslint-disable react/prop-types */

import React from 'react';
import { render } from '@testing-library/react';

import withTheme from '../../../../themes/withThemeProvider';
import withIntl from '../../../../i18n/withIntlProvider';
import MakeClaimHeader from '../MakeClaimHeader';

jest.mock('../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

jest.mock(
  '../../../../uiComponents/Typography',
  () => ({ children, ...rest }) => <div {...rest}>{children}</div>,
);

describe('MakeClaimHeader Component', () => {
  it('should match the snapshot', () => {
    const Component = withIntl(withTheme(MakeClaimHeader));

    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
