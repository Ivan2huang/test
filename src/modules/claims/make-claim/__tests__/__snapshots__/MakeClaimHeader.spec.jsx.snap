// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeClaimHeader Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        type="style-1"
      >
        Make a claim
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        type="style-8"
      >
        Claim submission is only applicable to designated outpatient consultation incurred in HK (refer to ‘Consultation Type’ in the next step). Please keep your original receipt as HSBC Life BenefitsPlus reserves the right to request for the original receipt within 6 months of the claim completion date. For details, please refer to the leaflets.
      </div>
    </div>
  </div>
</div>
`;
